common/djangoapps/util/model_utils.py

```diff
@@ -180,7 +180,7 @@ def generate_unique_readable_id(name, queryset, lookup_field):
         string: generated unique identifier
     """
     candidate = slugify(name)
-    conflicts = queryset.filter(**{lookup_field + '__startswith': candidate}).values_list(lookup_field, flat=True)
+    conflicts = queryset.filter(**{lookup_field + '__istartswith': candidate}).values_list(lookup_field, flat=True)
```

Comment:
Replying inline. Say I enter the name "T", and there are 50K names that start with "t" (e.g. from "The ..."). If I'm reading that correctly, that means that we're going to pull across 50K rows across the wire from the database and load it into the conflicts list? We don't need a complex name generation algorithm -- I'd like a reasonable upper limit on what can come back here.